src/integration/java/io/pinecone/integration/dataPlane/UpdateFetchAndQueryPodTest.java

```diff
@@ -121,49 +115,50 @@ public void updateAllParamsFetchAndQuerySyncTest() throws InterruptedException {
                     null,
                     true,
                     true);
+            List<ScoredVectorWithUnsignedIndices> queryResults = queryResponse.getMatchesList();
+            ScoredVectorWithUnsignedIndices scoredVectorV1 = null;
+
+            for (ScoredVectorWithUnsignedIndices vector : queryResults) {
+                if (idToUpdate.equals(vector.getId())) {
+                    scoredVectorV1 = vector;
+                }
+            }
 
-            ScoredVectorWithUnsignedIndices scoredVectorV1 = queryResponse.getMatches(0);
             // Verify the correct vector id was updated
+            assertNotNull(scoredVectorV1);
             assertEquals(scoredVectorV1.getId(), idToUpdate);
 
             // Verify the updated values
-            List<Float> valuesList = new ArrayList<>(scoredVectorV1.getValuesList());
-            Collections.sort(valuesList);
-            Collections.sort(valuesToUpdate);
+            List<Float> valuesList = scoredVectorV1.getValuesList();
             assertEquals(valuesToUpdate, valuesList);
 
             // Verify the updated metadata
             assertEquals(scoredVectorV1.getMetadata(), metadataToUpdate);
 
             // Verify the initial sparse values set for upsert operation
-            List<Float> expectedSparseValues = new ArrayList<>(scoredVectorV1.getSparseValuesWithUnsignedIndices().getValuesList());
-            List<Float> actualSparseValues = new ArrayList<>(sparseValuesList.get(0));
-            Collections.sort(actualSparseValues);
-            Collections.sort(expectedSparseValues);
-
+            List<Float> expectedSparseValues = sparseValuesList;
+            List<Float> actualSparseValues = scoredVectorV1.getSparseValuesWithUnsignedIndices().getValuesList();
             assertEquals(expectedSparseValues, actualSparseValues);
-        });
+        }, 3);
```

Comment:
Is this asserting that both `expectedSparseValues` and `actualSparseValues` are equal to `3`?

Reply:
The 3 that's added here is for setting the `backOff` to 3 for `assertWithRetry`. The `assertEquals` above is asserting what you're describing though.